cliquet/resource/__init__.py

```diff
@@ -826,13 +827,9 @@ def _extract_fields(self):
                 }
                 raise_invalid(self.request, **error_details)
 
-        return fields
+            fields = fields + [self.model.id_field, self.model.modified_field]
```

Comment:
This is because the id and last_modified field should always be return as part as the synchronisation protocol.
If the user forgot to add them in the _fields list we still want them in the record object.